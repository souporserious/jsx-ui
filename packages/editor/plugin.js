const { createSextantPlugin } = require('@sextant-tools/core')
const {
  buildBaseTypeFiles,
} = require('@sextant-tools/plugin-javascript-operations')
const Handlebars = require('handlebars')
const fs = require('fs')
const path = require('path')

const plugin = createSextantPlugin(
  (
    context,
    { expressFileName = 'sextant-express.generated', typescriptFileName }
  ) => {
    // console.log(context, expressFileName, typescriptFileName)
    const files = buildBaseTypeFiles(context.database, typescriptFileName)

    files.forEach((file) => {
      context.writeFileSync(file.filename, file.content)
    })

    const declarationFileTemplate = Handlebars.compile(
      fs
        .readFileSync(path.resolve(__dirname, './express-plugin.d.ts.hbs'))
        .toString()
    )

    context.writeFileSync(
      expressFileName + '.d.ts',
      declarationFileTemplate({})
    )

    const jsFileTemplate = fs
      .readFileSync(path.resolve(__dirname, './express-plugin.js'))
      .toString()

    context.writeFileSync(expressFileName + '.js', jsFileTemplate)
  }
)

module.exports = plugin
