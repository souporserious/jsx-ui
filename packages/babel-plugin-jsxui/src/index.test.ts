import pluginTester from 'babel-plugin-tester/pure'

import plugin from '../src'

pluginTester({
  plugin,
  pluginName: 'babel-plugin-jsxui',
  filename: __filename,
  snapshot: true,
  tests: [{ fixture: '__fixtures__/simple.js' }],
})
