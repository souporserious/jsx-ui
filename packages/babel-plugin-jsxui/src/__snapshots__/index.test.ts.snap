// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-jsxui 1. babel-plugin-jsxui: 1. babel-plugin-jsxui 1`] = `
"
import { Overrides, Text } from 'jsxui'

function App() {
  return (
    <Overrides
      value={[
        <Text color=\\"tomato\\" variants={{ hover: { color: 'papayawhip' } }} />,
      ]}
    >
      <Text>Hello World</Text>
    </Overrides>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { Overrides, Text } from 'jsxui';

function App() {
  return <Overrides value={[[Text, {
    color: \\"tomato\\",
    variants: {
      hover: {
        color: 'papayawhip'
      }
    }
  }]]}>
      <Text>Hello World</Text>
    </Overrides>;
}
"
`;
